src/Language/PureScript/Ide/CaseSplit.hs

```diff
@@ -119,11 +120,17 @@ addClause s wca =
         " = ?" <> (T.strip . T.pack . runIdent $ fName)
   in [s, template]
 
-parseType' :: String -> Type
-parseType' s = let (Right t) = do
-                     ts <- lex "" s
-                     runTokenParser "" (parseType <* P.eof) ts
-               in t
+parseType' :: (MonadError PscIdeError m) =>
+              String -> m Type
+parseType' s = let
+                 parseResult = do
+                   ts <- lex "" s
+                   runTokenParser "" (parseType <* P.eof) ts
+               in
+                 case parseResult of
```

Comment:
`case lex "" s >>= runTokenParser "" (parseType <* P.eof) of ...` perhaps?

Also, consider using a filename instead of `""` since the filename shows up in errors now.